Sources/Datadog/Core/Persistence/FilesOrchestrator.swift

```diff
@@ -61,6 +61,10 @@ internal class FilesOrchestrator {
 
     private func reuseLastWritableFileIfPossible(writeSize: UInt64) -> WritableFile? {
         if let lastFileName = lastWritableFileName {
+            if !directory.hasFile(named: lastFileName) {
+                return nil // this is expected if the last writable file was deleted after upload
+            }
+
             do {
                 let lastFile = try directory.file(named: lastFileName)
```

Comment:
we can merge those 2 into `guard let lastFile = directory.file(named:) else { return nil }`
that's not a major issue but currently we catch `try directory.file(named:)` and `try lastFile.size()` in the same block, so we can't know which error happened when

does that make sense?